src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -971,4 +969,35 @@ static boolean upgradePrefs(Context context, String sourcePkgName, String target
             Log.e(TAG, "Error upgrading shared preferences", e);
             return false;
         }
-    }}
+    }
+
+    /*
+     * Move device ID from sharedPrefs to new sqlite key value store.
+     *
+     * This should only happen once -- the first time a user loads the app after updating.
+     * This should happen only after moving the preference data from legacy to new static name.
+     * This logic needs to remain in place for quite a long time. It was first introduced in
+     * August 2015 in version 1.8.0.
+     */
+    static boolean upgradeDeviceIdToDB(Context context) {
+        return upgradeDeviceIdToDB(context, null);
+    }
+
+    static boolean upgradeDeviceIdToDB(Context context, String sourcePkgName) {
+        if (sourcePkgName == null) {
+            sourcePkgName = Constants.PACKAGE_NAME;
+        }
+
+        String prefsName = sourcePkgName + "." + context.getPackageName();
+        SharedPreferences preferences =
+                context.getSharedPreferences(prefsName, Context.MODE_PRIVATE);
+
+        String deviceId = preferences.getString(Constants.PREFKEY_DEVICE_ID, null);
+        if (!TextUtils.isEmpty(deviceId)) {
+            DatabaseHelper dbHelper = DatabaseHelper.getDatabaseHelper(context);
+            dbHelper.insertOrReplaceKeyValue(DEVICE_ID_KEY, deviceId);
+        }
```

Comment:
you should remove the source device id afterwards so this transfer only happens once.